deeplearning4j/deeplearning4j-nn/src/main/java/org/deeplearning4j/nn/conf/InputPreProcessor.java

```
@@ -69,4 +76,12 @@
 
 
     Pair<INDArray, MaskState> feedForwardMaskArray(INDArray maskArray, MaskState currentMaskState, int minibatchSize);
+
+    /**
+     * Define the InputPreProcessor's input transformation in a {@link SameDiff} instance.
```

Maybe we add: "Custom preprocessors may throw an UnsupportedOperationException if the DL4J net will never be converted to SameDiff, or if SameDiff conversion is not supported" (or something to that effect)